### PR TITLE
Refactor IPA users and groups, make the data structure make sense

### PR DESCRIPTION
**Pull request by lhellebr**

### CODE REVIEW

**Comment by lhellebr** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py -k test_userlist_with_external_admin -k test_verify_group_permissions

**Comment by pnovotny** [comment]
PRT failed with:
```
dynaconf.validator.ValidationError: ipa.users is required in env main
```

**Comment by lhellebr** [comment]
Yes, I'm waiting for a config PR merge

**Comment by lhellebr** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py -k test_userlist_with_external_admin -k test_verify_group_permissions

**Comment by lhellebr** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py -k test_userlist_with_external_admin -k test_verify_group_permissions
env:
  ROBOTTELO_fam__server__external_usergroup_name: satadmins
  ROBOTTELO_ipa__users__user: satuser_01
  ROBOTTELO_ipa__users__admin: satadmin_01
  ROBOTTELO_ipa__groups__users: satusers
  ROBOTTELO_ipa__groups__admin: satadmins
```

**Comment by lhellebr** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py -k test_userlist_with_external_admin -k test_verify_group_permissions
env:
  ROBOTTELO_fam__server__external_usergroup_name: satadmins
  ROBOTTELO_ipa__users__user: satuser_01
  ROBOTTELO_ipa__users__admin: satadmin_01
  ROBOTTELO_ipa__groups__users: satusers
  ROBOTTELO_ipa__groups__admins: satadmins
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9040
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_ldap_authentication.py -k test_userlist_with_external_admin -k test_verify_group_permissions --external-logging
Test Result : ========== 1 passed, 53 deselected, 62 warnings in 677.19s (0:11:17) ===========
```

**Review by pnovotny — APPROVED** [review]
LGTM

File: tests/foreman/ui/test_ldap_authentication.py [hunk 1]
```diff
@@ -103,16 +103,6 @@ def rhsso_groups_teardown(default_sso_host):
...
 
 
-@pytest.fixture
-def multigroup_setting_cleanup(default_ipa_host):
-    """Adding and removing the user to/from ipa group"""
-    sat_users = settings.ipa.groups
-    idm_users = settings.ipa.group_users
-    default_ipa_host.add_user_to_usergroup(idm_users[1], sat_users[0])
-    yield
-    default_ipa_host.remove_user_from_usergroup(idm_users[1], sat_users[0])
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ldap_authentication.py:113]: Isn't it required now to add `satuser_01` in the `satadmin` group? If not, would you mind describing why?

Comment (lhellebr) [reply tests/foreman/ui/test_ldap_authentication.py:113]: User `satadmin` is in `satusers` group, that is stable and doesn't need to be done everytime.
Although you made me realise that one of the users in tests should actually be `satadmin` for the test to cover what it should cover, so thanks for this question!
Also, I will rerun PRT.

**Comment by lhellebr** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_ldap_authentication.py -k test_userlist_with_external_admin -k test_verify_group_permissions
env:
  ROBOTTELO_fam__server__external_usergroup_name: satadmins
  ROBOTTELO_ipa__users__user: satuser_01
  ROBOTTELO_ipa__users__admin: satadmin_01
  ROBOTTELO_ipa__groups__users: satusers
  ROBOTTELO_ipa__groups__admins: satadmins
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9188
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_ldap_authentication.py -k test_userlist_with_external_admin -k test_verify_group_permissions --external-logging
Test Result : ========== 1 passed, 53 deselected, 62 warnings in 625.42s (0:10:25) ===========
```

**Review by pondrejk — APPROVED** [review]

**Comment by lhellebr** [comment]
This has been open since september, can someone please click the green button?

**Comment by damoore044** [comment]
@lhellebr With these newly merged changes, locally (6.16.0) I am now seeing at start of test session 
`dynaconf.validator.ValidationError: ipa.users is required in env main`
[Full Stacktrace](https://privatebin.corp.redhat.com/?e500a16257149329#HD4fyMnuRDQ4R1Rbrz2CjUaA3qE74WHtTcfFUFVQkFjg) , dropping the commit fixed locally for me. Any ideas?

**Comment by lhellebr** [comment]
@damoore044 do you have your config up to date? Config PR 1501.

**Pull request merged**